### PR TITLE
Sprint 1 changes

### PR DESCRIPTION
**Pull request by darrellcolehill**
Initial development of the app. 

Components to be examined 
- Text.tsx: processes and displays OSIS compliant ULB XML 
- TextView.tsx: container for desktop navigation icons, Text, and ChapterNavigationBar
- ChapterNavigationBar.tsx: component for navigation bar
- App.tsx: container for TextView and GreekWordsDialog
- GreekWordsDialog.tsx: container for GreekWordsInfo
- GreekWordsinfo.tsx: displays data for given Greek word.
- NextFab.tsx: next chapter floating action button for mobile devices 
- PreviousFab.tsx: previous chapter floating action button for mobile devices 
- StationaryChapterNavButton.tsx: chapter navigation button for large devices

dummy-backend/convertToOSIS.js: takes the non-OSIS compliant book of Philemon (downloaded from the en_ULB_tagged repo) and generates an XML file containing a semi-OSIS compliant version of Philemon and a JSON version of the newly created XML.

### CODE REVIEW

File: src/components/TextView.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import Grid from '@mui/material/Grid';
+import ChapterNavigationBar from './ChapterNavigationBar';
+import StationaryChapterNavButton from './StationaryChapterNavButton';
+import ArrowRightAltIcon from '@material-ui/icons/ArrowForward';
+import ArrowLeftAltIcon from '@material-ui/icons/ArrowBack';
+import Text from './Text';
+import react, {useEffect, useState} from 'react';
+
+
+type Note = {
```
Comment (mXaln) [inline src/components/TextView.tsx]: You could import types from types.ts, instead of creating duplicate type here

File: src/components/Text.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
+import { CollectionsBookmarkRounded } from '@material-ui/icons';
+import React, { useState, useEffect, useRef } from 'react';
+import getChapter from '../api';
+import "../App.css";
+
+
+type Note = {
```
Comment (mXaln) [inline src/components/Text.tsx]: Same as above. Types could be imported from types.ts file

File: src/App.tsx [hunk 3]
```diff
@@ -37,221 +26,39 @@ const Item : any = styled(Paper)(({ theme }) => ({
 }));
 
 
-function ChapterNavigationBar()
-{
-  return (
-
-    <Grid container direction="row" justifyContent={{lg: "center", md: "center", sm:"flex-start"}} alignItems="flex-start" style={{marginTop:"15px", marginLeft:"15px", marginBottom:"15px", height:"50px"}}>
```
Comment (mXaln) [inline src/App.tsx:44]: I wouldn't use inline styles, cause they are really long and repetitive. I would recommend to use class names

File: dummy-backend/convertToOSIS.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    const builder = new XMLBuilder(orderedParserOptions);
+    const xmlOutput = builder.build(orderedXML);
+
+    return xmlOutput;
+
+}
+
+
+// Converts current ULB XML file to an OSIS complient ULB XML file and then converts that to JSON
+let newXML = taggedULBToOsisULB("testXML/58-PHM.xml");
```
Comment (PurpleGuitar) [inline dummy-backend/convertToOSIS.js:142]: This should probably some kind of config rather than hard-coded.

File: src/components/NextFab.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import Fab from '@mui/material/Fab';
```
Comment (PurpleGuitar) [inline src/components/NextFab.tsx]: I think NextFab and PreviousFab should be named according to their purpose, not their implementation.  Although they happen to be floating action buttons now, they might have a different implementation in the future.  Likewise, other developers might not know what "NextFab" is for, but "NextChapterButton" is crystal-clear.

File: src/components/PreviousFab.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import Fab from '@mui/material/Fab';
```
Comment (PurpleGuitar) [inline src/components/PreviousFab.tsx]: Please see comment for NextFab.

File: src/api/index.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { cp } from "fs";
+let PHMJSON = require('./testData/PHM.json');
```
Comment (PurpleGuitar) [inline src/api/index.ts:1]: Unless this is part of a test suite, it shouldn't be hard coded.

File: src/components/GreekWordsDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+type greekWord = {
+  greekWords: string,
+  englishWords: string,
+  morphology: string,
+  descriptions: string[],
+};
+
+let word1 : greekWord = {
+  greekWords: "asdkfjasdgjkb/asdfasdf",
```
Comment (PurpleGuitar) [inline src/components/GreekWordsDialog.tsx]: I guess this is placeholder text?

File: src/types.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
+type Note = {
+    ATTR: any,
+    _: any,
+}
+  
+type w = {
```
Comment (PurpleGuitar) [inline src/types.ts]: "w" is not a good name for an exported variable.  Does it need to be "w" for JSON importing, or some other reason?  If not, then I recommend naming this something that will have meaning for other developers, like "word".

File: src/components/TextView.tsx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+  ATTR: any,
+  _: any,
+}
+
+type w = {
```
Comment (PurpleGuitar) [inline src/components/TextView.tsx]: See comment in types.ts re: a variable called "w".

File: src/components/TextView.tsx [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+function TextView({size, setSize, setCurrentGreekWord}: TextViewProps)
+{
+
+  const onPhraseClick = (greekWord : GreekWord[]) =>
+  {
+    setCurrentGreekWord(greekWord);
+
+    if(size === 12)
```
Comment (PurpleGuitar) [inline src/components/TextView.tsx:24]: What is the purpose of this transformation?  I recommend adding a comment for tricky code like this.

File: src/components/Text.tsx [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+  ATTR: any,
+  _: any,
+}
+
+type w = {
```
Comment (PurpleGuitar) [inline src/components/Text.tsx]: See comment in types.ts re: a variable called "w".

File: src/components/Text.tsx [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+interface TextProps {
+    onPhraseClick: (greekWordNotes : GreekWord[]) => any;
+}
+
+function Text({onPhraseClick}: TextProps)
+{
+  const [verses, setVerses] = useState([] as verse[]);
+
+  useEffect(() => {
+    let data = getChapter()
```
Comment (PurpleGuitar) [inline src/components/Text.tsx]: Missing semicolon?

File: src/components/Text.tsx [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+          
+          if(hasConsumedSubWord)// buffer contain subWords. NOTE: Current word is processed in this case
+          {
+            currentGreekWord.text = currentGreekWord.text.replace('[1]', greekWords[0].text);
+            greekWords.push(currentGreekWord)
+            verseWordOutput.push(<><span className="GreekPhrase" onClick={() => {onPhraseClick([...greekWords])}}>{currentGreekWord.text}</span><span> </span></>)
+            hasConsumedSubWord = false;
+            greekWordBuffer.splice(0, greekWordBuffer.length);
+          }
+          else if(word._ !== "√") // current word has english backing
```
Comment (PurpleGuitar) [inline src/components/Text.tsx]: What is the underscore in this context?  It looks like a string that contains a checkmark unicode character?  Could this be a boolean?

**Review by PurpleGuitar — APPROVED** [review]

File: src/components/GreekWordsDialog.tsx (hunk 8 above)
Comment (darrellcolehill) [reply src/components/GreekWordsDialog.tsx]: yes, I am currently using it to test the design that will be used to show the data from gwt

**Review by mXaln — CHANGES_REQUESTED** [review]

File: src/components/Text.tsx (hunk 14 above)
Comment (darrellcolehill) [reply src/components/Text.tsx]: Symbol indicates that the word has no English meaning

**Pull request merged**